apps/api/src/app/events/services/workflow-queue/workflow.queue.service.ts

```diff
@@ -32,40 +40,45 @@ export class WorkflowQueueService {
   public readonly queue: Queue;
   public readonly worker: Worker;
   @Inject()
-  private jobRepository: JobRepository;
+  private createExecutionDetails: CreateExecutionDetails;
+  @Inject()
+  private queueNextJob: QueueNextJob;
   @Inject()
   private runJob: RunJob;
+  @Inject()
+  private setJobAsCompleted: SetJobAsCompleted;
+  @Inject()
+  private setJobAsFailed: SetJobAsFailed;
```

Comment:
Would suggest moving all of those to the constructor with @Inject, easier to read in my opinion and follows the pattern of all our other services

Reply:
Doing this way we are forced to use the `forwardRef` functionality as injecting them as arguments generates a circular dependency with the Events module due `AddJob` depends on the Workflow Queue service.